runtime/lua/vim/shared.lua

```diff
@@ -188,6 +188,31 @@ local function pesc(s)
   return s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1')
 end
 
+--- Test if `prefix` is a prefix of `s`
+--
+-- @tparam string s String to check
+-- @tparam string prefix Potential prefix
+-- @treturn boolean True if prefix is a prefix of s
+local function startswith(s, prefix)
+  assert(type(s) == 'string', 'Input must be a string')
+  assert(type(prefix) == 'string', 'Prefix must be a string')
+	return s:sub(1, #prefix) == prefix
+end
+
+--- Test if `suffix` is a suffix of `s`
+--
+-- @tparam string s String to check
+-- @tparam string suffix Potential suffix
+-- @treturn boolean True if suffix is a suffix of s
+local function endswith(s, suffix)
+  assert(type(s) == 'string', 'Input must be a string')
+  assert(type(suffix) == 'string', 'Suffix must be a string')
```

Comment:
```suggestion
  assert(type(suffix) == 'string', 'expected string, got '..type(suffix))
```

Maybe we should introduce a `validate` module so we can have consistent messages.

    validate.is_type('string', 'suffix', suffix)

Reply:
Metalua is very interesting. But unfortunately, maintenance has stopped and it has been many years.
The lua-resty-validation looks good, but it seems a bit too big to introduce just for this purpose.
Can I make a function or module for type assertion in another PR and discuss it there?

Reply:
> Can I make a function or module for type assertion in another PR and discuss it there?

👍

Reply:
current master now has `vim.validate()`. 

Example: https://github.com/neovim/neovim/commit/0190de9aab53c0b82e648c51c5162d6377c6f341#diff-7f472431a1463bbeec031cd66114173eR106-R108